enhancements/authentication/direct-external-oidc-provider.md

```diff
@@ -0,0 +1,409 @@
+
+#### External OIDC provider configuration
+
+The configuration details required to connect to an external OIDC provider are:
+- the provider URL
+- the ID and secret of the corresponding OIDC client at the provider's side
+- its certificate authority bundle, if applicable
+- any relevant extra scopes, if applicable
+
+#### Authentication CR
+
+The cluster's Authentication CR (named `cluster`) must be modified and updated with the configuration of the external OIDC provider in the `OIDCProviders` field.
+
+Once the CR gets updated, the changes will be picked up automatically by the cluster-kube-apiserver-operator, the cluster-authentication-operator and the console-operator; the operators will then update their operands accordingly and will deactivate/remove all relevant controllers/resources.
+
+For more information on the Authentication CR API, see [API Extensions](#api-extensions).
+
+#### `oc` CLI tool
+
+`oc` supports the specification of the details of an external OIDC provider to be used via an exec plugin (currently, only `oc-oidc` is supported). See `oc login --help` for more details on how to specify these details.
+
+Note that the necessary changes have already been implemented in `oc`; see [oc#1640](https://github.com/openshift/oc/pull/1640).
+
+### API Extensions
+
+To facilitate the specification of the external OIDC provider configuration, the Authentication CRD is extended with a new field `OIDCProviders`, in its spec:
+```go
+type AuthenticationSpec struct {
+	...
+
+	// OIDCProviders are OIDC identity providers that can issue tokens
+	// for this cluster
+	// Can only be set if "Type" is set to "OIDC".
+	//
+	// At most one provider can be configured.
+	//
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=1
+	// +openshift:enable:FeatureGate=ExternalOIDC
+	OIDCProviders []OIDCProvider `json:"oidcProviders,omitempty"`
+}
+```
+
+For more details on the `OIDCProvider` type and its fields, see [here](https://github.com/openshift/api/blob/master/config/v1/types_authentication.go#L197).
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This enhancement proposal is not relevant to Hypershift; this has been implemented independently for Hypershift (see [OCPSTRAT-933](https://issues.redhat.com/browse/OCPSTRAT-933))
+
+#### Standalone Clusters
+
+This enhancement proposal applies to standalone OCP.
+
+*TODO: what else to write here?*
+
+#### Single-node Deployments or MicroShift
+
+Configuring an external OIDC provider in a Single-Node deployment of OpenShift (SNO) will result in reduced resource consumption overall, due to the fact that once an external provider is configured successfully, the system will deactivate/remove components and resources that are unused (e.g. the oauth-server and oauth-apiserver pods won't run anymore).
```

Comment:
AFAIK MicroShift doesn't run with multiple users and it does not even deploy the integrated oauth-server